contracts/extensions/ERC721AQueries.sol

```diff
@@ -0,0 +1,156 @@
+     *
+     * If the `tokenId` is burned:
+     *   - `addr` = `<Address of owner before token was burned>`
+     *   - `startTimestamp` = `<Timestamp when token was burned>`
+     *   - `burned = `true`
+     *
+     * Otherwise:
+     *   - `addr` = `<Address of owner>`
+     *   - `startTimestamp` = `<Timestamp of start of ownership>`
+     *   - `burned = `false`
+     */
+    function rawOwnershipOf(uint256 tokenId) public view returns (TokenOwnership memory) {
+        if (_exists(tokenId)) {
+            return _ownershipOf(tokenId);
+        }
+        TokenOwnership memory ownership;
+        if (_ownerships[tokenId].burned) {
+            ownership = _ownerships[tokenId];
+        }
+        return ownership;
+    }
+
+    /**
+     * @dev Returns an array of `TokenOwnership` structs at `tokenIds` in order.
+     * See {ERC721AQueries-rawOwnershipOf}
+     */
+    function rawOwnershipsOf(uint256[] memory tokenIds) external view returns (TokenOwnership[] memory) {
+        unchecked {
+            uint256 n = tokenIds.length;
+            TokenOwnership[] memory a = new TokenOwnership[](n);
+            for (uint256 i; i != n; ++i) {
+                a[i] = rawOwnershipOf(tokenIds[i]);
+            }
+            return a;
+        }
+    }
+
+    /**
+     * @dev Returns an array of token IDs owned by `owner`,
+     * in the range [`start`, `stop`)
+     * (i.e. `start <= tokenId < stop`).
+     *
+     * This function allows for tokens to be queried if the collection
+     * grows too big for a single call of {ERC721AQueries-tokensOfOwner}.
+     */
+    function tokensOfOwnerIn(
+        address owner,
+        uint256 start,
+        uint256 stop
+    ) external view returns (uint256[] memory) {
+        unchecked {
+            uint256 tokenIdsIdx;
+            uint256 stopLimit = _currentIndex;
+            // Set `start = max(start, _startTokenId())`.
+            if (start < _startTokenId()) {
+                start = _startTokenId();
+            }
+            // Set `stop = min(stop, _currentIndex)`.
+            if (stop > stopLimit) {
+                stop = stopLimit;
+            }
+            uint256 n = balanceOf(owner);
+            // Set `n = min(balanceOf(owner), stop - start)`,
+            // to cater for cases where `balanceOf(owner)` is too big.
+            if (start < stop) {
+                uint256 d = stop - start;
+                if (d < n) {
+                    n = d;
+                }
+            } else {
+                n = 0;
+            }
+            uint256[] memory a = new uint256[](n);
+            // We need to call `rawOwnershipOf(start)`, 
+            // because the slot at `start` may not be initialized.
+            TokenOwnership memory ownership = rawOwnershipOf(start);
+            address currOwnershipAddr;
+            // If the starting slot exists, initialize `currOwnershipAddr`.
+            if (ownership.addr != address(0) && !ownership.burned) {
```

Comment:
What happens when this statement isn't true? Do we actually need to initialize `currOwnershipAddr`?

Reply:
When this statement isn't true, it means that the scanning window starts on a burned slot, which should be skipped in the loop.

We need to initialize `currOwnershipAddr`, because the scanning window can start in the middle on a batch that has uninitialized slots.